pkg/parser/external/java_parser_server.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package external
+
+import (
+	"fmt"
+	"google.golang.org/grpc"
+	"log"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+var (
+	mu         sync.Mutex
+	clientConn *grpc.ClientConn
+)
+
+func getServerPort() string {
+	port := os.Getenv("SQLFLOW_PARSER_SERVER_PORT")
+	if port == "" {
+		log.Fatal("undefined environment variable SQLFLOW_PARSER_SERVER_PORT")
+	}
+	return port
+}
+
+func isServerUp() bool {
+	cmd := exec.Command("curl", "-v", fmt.Sprintf("localhost:%s", getServerPort()))
```

Comment:
Can we do not assume the existence of curl as a dependency?

```go
// Below init function
func TestNETServer_Run(t *testing.T) {
    // Simply check that the server is up and can
    // accept connections.
    conn, err := net.Dial("tcp", ":1123")
    if err != nil {
        t.Error("could not connect to server: ", err)
    }
    defer conn.Close()
}
```

Reply:
I will try out `net.Dial`. I previous attempt on `grpc.Dial` gives nil error while the server doesn't exist.